pkg/util/fxutil/args.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fxutil
+
+import (
+	"reflect"
+
+	"go.uber.org/fx"
+)
+
+var errorInterface = reflect.TypeOf((*error)(nil)).Elem()
+
+// delayedFxInvocation delays execution of a function, while allowing
+// Fx to provide its arguments.
+type delayedFxInvocation struct {
+	fn    interface{}
+	ftype reflect.Type
+	args  []reflect.Value
+}
+
+// newDelayedFxInvocation creates a new delayedFxInvocation wrapping the given
+// function.
+//
+// The given function can have any number of arguments that will be supplied with Fx.
+// It must return nothing or an error.
+func newDelayedFxInvocation(fn interface{}) *delayedFxInvocation {
+	ftype := reflect.TypeOf(fn)
+	if ftype == nil || ftype.Kind() != reflect.Func {
+		panic("delayedFxInvocation requires a function as its first argument")
+	}
+
+	// verify it returns error
+	if ftype.NumOut() > 1 || (ftype.NumOut() == 1 && !ftype.Out(0).Implements(errorInterface)) {
```

Comment:
```suggestion
	if ftype.NumOut() == 0 || (ftype.NumOut() == 1 && !ftype.Out(0).Implements(errorInterface)) {
```
?

Reply:
Oh, sorry, this was correct as-is.  It's allowed to return nothing, but not >1 return value.